mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp

```diff
@@ -984,6 +985,31 @@ class AvgPool2dConverter : public OpRewritePattern<tosa::AvgPool2dOp> {
   }
 };
 
+class TransposeConverter : public OpRewritePattern<tosa::TransposeOp> {
+public:
+  using OpRewritePattern<tosa::TransposeOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(tosa::TransposeOp op,
+                                PatternRewriter &rewriter) const final {
+    SmallVector<int64_t> constantPerms;
+    if (failed(op.getConstantPerms(constantPerms)))
```

Comment:
Is there any reason to not delete the existing lowering in `TosaToLinalg` as well? This rewrite looks like it supports the same set of cases, and I think it is required to run `TosaToLinalgNamed` to `TosaToLinalg`, so the old lowering would effectively be dead code.

Reply:
> I think it is required to run `TosaToLinalgNamed` to `TosaToLinalg`

I wasn't aware of that requirement and kept the old Pattern mostly in case any downstream pipelines were relying on the direct `tosa.transpose`->`linalg.generic` pathway. But I agree, the two Patterns support the same cases and the old one can be deleted.

Reply:
I went ahead and deleted the old pattern, moving all related tests from `tosa-to-linalg.mlir` to `tosa-to-linalg-named.mlir`